src/ext/engine_hooks.c

```diff
@@ -41,12 +42,14 @@ static zend_op_array *_dd_compile_file(zend_file_handle *file_handle, int type T
 }
 
 static void _compile_minit(void) {
-    _prev_compile_file = zend_compile_file;
-    zend_compile_file = _dd_compile_file;
+    if (get_dd_trace_measure_compile_time()) {
```

Comment:
I'm wondering if we should keep the `zend_compile_file` hook, but just feature-flag the sys calls. As it stands right now, a user will have to restart FPM/Apache after changing `DD_TRACE_MEASURE_COMPILE_TIME`.

Reply:
That's a fair point. At the same time, having the hook will still take longer than not at all; for customers experiencing SIGBUS minimizing this time is important.

Reply:
What if we flag the inside and outside? This way customers moving from having it ON (default) can move back and forth until restart, but after starting with it OFF it will not have a meaningful effect until restart (will report 0 time for all compiles).

Reply:
I like it! :)

Reply:
The check turns out to do a lock/unlock, so it won't fair any better performance wise by doing it on the inside. So I think as written is the best option.

I think this just needs a test and it should be good to go.